src/providers/mssql/qgsmssqlprovider.cpp

```diff
@@ -947,10 +952,14 @@ bool QgsMssqlProvider::addFeatures( QgsFeatureList &flist, Flags flags )
     statement += QStringLiteral( ") " );
     if ( !( flags & QgsFeatureSink::FastInsert ) )
     {
-      statement += QStringLiteral( " OUTPUT inserted." ) + mFidColName;
+      statement += QStringLiteral( " OUTPUT inserted." ) + mFidColName + QStringLiteral( "  INTO @px " );
```

Comment:
I don't know mysql very well. Should inserted and the column name be quoted (something like "inserted"."colname")?

Reply:
Quotations are not necessary but MSSQL likes when databases schema and tables are between bracket but the inserted table can be referenced without quotation and other such.

Here is an example with OUTPUT https://docs.microsoft.com/en-us/sql/t-sql/queries/output-clause-transact-sql?view=sql-server-2017#examples

Reply:
Actually I think there should be a `QgsMssqlProvider::quotedIdentifier` method doing the same thing as the [`QUOTENAME`](https://docs.microsoft.com/en-us/sql/t-sql/functions/quotename-transact-sql?view=sql-server-2017) function. But that's out of scope for this pull request.